File: deps/yeppp-1.0.0/examples/java/sources/CpuInfo.java
import java.util.Iterator;

class CpuInfo {
	
	public static void main(String[] args) {
		final info.yeppp.CpuArchitecture architecture = info.yeppp.Library.getCpuArchitecture();
		final info.yeppp.CpuVendor vendor = info.yeppp.Library.getCpuVendor();
		final info.yeppp.CpuMicroarchitecture microarchitecture = info.yeppp.Library.getCpuMicroarchitecture();
		System.out.println("Architecture: " + architecture.getDescription());
		System.out.println("Vendor: " + vendor.getDescription());
		System.out.println("Microarchitecture: " + microarchitecture.getDescription());
		final Iterator<info.yeppp.CpuIsaFeature> isaFeaturesIterator = architecture.iterateIsaFeatures();
		if (isaFeaturesIterator.hasNext()) {
			System.out.println("CPU ISA extensions:");
			while (isaFeaturesIterator.hasNext()) {
				final info.yeppp.CpuIsaFeature isaFeature = isaFeaturesIterator.next();
				System.out.println(String.format("\t%-60s\t%s", isaFeature.getDescription() + ":", (info.yeppp.Library.isSupported(isaFeature) ? "Yes" : "No")));
			}
		}
		final Iterator<info.yeppp.CpuSimdFeature> simdFeaturesIterator = architecture.iterateSimdFeatures();
		if (simdFeaturesIterator.hasNext()) {
			System.out.println("CPU SIMD extensions:");
			while (simdFeaturesIterator.hasNext()) {
				final info.yeppp.CpuSimdFeature simdFeature = simdFeaturesIterator.next();
				System.out.println(String.format("\t%-60s\t%s", simdFeature.getDescription() + ":", (info.yeppp.Library.isSupported(simdFeature) ? "Yes" : "No")));
			}
		}
		final Iterator<info.yeppp.CpuSystemFeature> systemFeaturesIterator = architecture.iterateSystemFeatures();
		if (systemFeaturesIterator.hasNext()) {
			System.out.println("Non-ISA CPU and system features:");
			while (systemFeaturesIterator.hasNext()) {
				final info.yeppp.CpuSystemFeature systemFeature = systemFeaturesIterator.next();
				System.out.println(String.format("\t%-60s\t%s", systemFeature.getDescription() + ":", (info.yeppp.Library.isSupported(systemFeature) ? "Yes" : "No")));
			}
		}
	}
	
}

File: deps/yeppp-1.0.0/examples/csharp/sources/CpuInfo.cs
using System;

class CpuInfo {

	public static void Main(string[] args)
	{
		/* Retrieve information about processor architecture */
		Yeppp.CpuArchitecture architecture = Yeppp.Library.GetCpuArchitecture();
		Console.WriteLine("Architecture: {0}", architecture.Description);
		/* Retrieve information about processor vendor */
		Yeppp.CpuVendor vendor = Yeppp.Library.GetCpuVendor();
		Console.WriteLine("Vendor: {0}", vendor.Description);
		/* Retrieve information about processor microarchitecture */
		Yeppp.CpuMicroarchitecture microarchitecture = Yeppp.Library.GetCpuMicroarchitecture();
		Console.WriteLine("Microarchitecture: {0}", microarchitecture.Description);
		/* Check if Yeppp! is aware of any ISA features on this architecture */
		if (architecture.CpuIsaFeatures.GetEnumerator().MoveNext()) {
			Console.WriteLine("CPU ISA extensions:");
			/* Iterate through ISA features */
			foreach (Yeppp.CpuIsaFeature isaFeature in architecture.CpuIsaFeatures)
			{
				Console.WriteLine(String.Format("\t{0, -60}\t{1}", isaFeature.Description + ":", (Yeppp.Library.IsSupported(isaFeature) ? "Yes" : "No")));
			}
		}
		/* Check if Yeppp! is aware of any SIMD features on this architecture */
		if (architecture.CpuSimdFeatures.GetEnumerator().MoveNext()) {
			Console.WriteLine("CPU SIMD extensions:");
			/* Iterate through SIMD features */
			foreach (Yeppp.CpuSimdFeature simdFeature in architecture.CpuSimdFeatures)
			{
				Console.WriteLine(String.Format("\t{0, -60}\t{1}", simdFeature.Description + ":", (Yeppp.Library.IsSupported(simdFeature) ? "Yes" : "No")));
			}
		}
		/* Check if Yeppp! is aware of any non-ISA CPU and system features on this architecture */
		if (architecture.CpuSystemFeatures.GetEnumerator().MoveNext()) {
			Console.WriteLine("Non-ISA CPU and system features:");
			/* Iterate through non-ISA CPU and system features */
			foreach (Yeppp.CpuSystemFeature systemFeature in architecture.CpuSystemFeatures)
			{
				Console.WriteLine(String.Format("\t{0, -60}\t{1}", systemFeature.Description + ":", (Yeppp.Library.IsSupported(systemFeature) ? "Yes" : "No")));
			}
		}
	}

}
